ui/src/components/BottomSheet.vue

```diff
@@ -7,14 +7,62 @@
       class="text-center"
       :height="windowHeight-100"
     >
-      <v-btn
-        class="mt-6"
-        text
-        color="error"
-        @click="show = !show"
-      >
-        close
-      </v-btn>
+      <v-card class="px-0 py-5">
+        <v-card-actions v-if="isCloseButtonShown">
+          <v-row
+            row
+            d-flex
+            nowrap
+            justify="end"
+            class="pa-4"
+          >
+            <v-btn
+              icon
+              text
+              small
+              fab
+              @click.stop="show=false"
+            >
+              <v-icon>mdi-close</v-icon>
+            </v-btn>
+          </v-row>
+        </v-card-actions>
+        <!-- use slot if default slot is given, else use default template -->
```

Comment:
I see that you copied the card element from the Modal to the BottomSheet. However, this is not necessary; rather than that, it should be the modal component that should be refactored because it is dirty XD   please discard this change and use modal for both views, should you decide not to implement the "confirm by entering the pwd one last time" feature, because if there is no pwd to submit, then using bottomsheet will be unnecessary.